### ISSUE TITLE
`@stylistic/string-quotes` rewrites the text of a `//` comment inside `<style lang="less">`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#101]
Found while fixing #32. The repair made there never reaches a stylesheet embedded in HTML, and the rule goes on rewriting the text of a comment — the one thing it must not do.

`string-quotes` decides whether a double slash opens a comment by asking the syntax: it hands one a probe stylesheet and reads what comes back. `postcss-html` is handed CSS rather than HTML, returns a `Document` with no roots at all, and so answers no — whatever language the stylesheet inside the page is actually written in. One syntax object serves `lang="less"`, `lang="scss"` and a plain `<style>` alike, so there is no answer it could give that would be right for all three.

### What minimal example or steps are needed to reproduce the bug?

```html
<style lang="less">
a {
  color: 'bar', // Some "comment"
    'baz';
}
</style>
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-html",
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/string-quotes": "single"
  }
}
```

### What did you expect to happen?

Nothing is reported. The quotation marks belong to the text of a comment, and the same stylesheet under `postcss-less` is passed over since #32 was fixed.

### What actually happened?

`3:25 Expected single quotes`, and `--fix` writes `// Some 'comment'` — it edits the comment. The behaviour is unchanged from before #32 was fixed, so this is a gap in the repair rather than a regression, but *Required behaviour* of that fix states plainly that the fix never rewrites a quotation mark sitting inside a comment.

`lang="scss"` is quiet, but by way of the `raws[prop].scss` fence of #100 rather than of anything done here, so it will need looking at when that fence is lifted.

### Do you have a proposal to fix the bug?

The question the rule asks is the right one; it is asked of the wrong thing. A `Document` hands each of its roots to an inner syntax, and it is that syntax, per root, which knows how it spells a comment. Either reach it, or put the question to the root that is actually being walked instead of to the syntax the whole file was opened with.
